src/main/java/taxi/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package taxi.service.impl;
+
+import java.util.Optional;
+import taxi.dao.DriverDao;
+import taxi.exception.AuthenticationException;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Driver;
+import taxi.service.AuthenticationService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverDao driverDao;
```

Comment:
I don't understand why and how. Because in DriverService I don't have the findByLogin method, only the get method but it get the driver by id, not by login